homeassistant/components/ecobee/binary_sensor.py

```diff
@@ -43,6 +43,16 @@ def name(self):
         """Return the name of the Ecobee sensor."""
         return self._name.rstrip()
 
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this sensor."""
+        for sensor in self.data.ecobee.get_remote_sensors(self.index):
+            if sensor["name"] == self.sensor_name:
+                if "code" in sensor:
```

Comment:
https://www.ecobee.com/home/developer/api/documentation/v1/objects/RemoteSensor.shtml

`id` seems to always be unique. Any reason why you would prefer `code` over `id` ?

Reply:
`id` seems to be unique based to the particular thermostat installation, and seems to be assigned sequentially based on when the sensor is added to the thermostat. `code` is printed on the actual sensor itself, so appears to be a better unique_id since the sensor could be paired to a different/new thermostat but would still identify itself to Home Assistant by its own unique identifier.

Sensors that do not have the 'code' property are built into the thermostat itself. Now that I think about it, in that case it actually may be better to use the thermostat's serial number suffixed with the device class ("-occupancy", "-humidity", or "-temperature") as a truly unique id that won't change based on the installation.

Reply:
Thinking about this further, I think it is best to use `id` for those sensors that are built into the thermostat - given that they are built in I think that the id assigned remains constant so it is a good unique identifier. My stance on using `code` for the remote sensors remains the same.

Reply:
ok!